src/gui/widgets/LcdFloatSpinBox.cpp

```diff
@@ -0,0 +1,243 @@
+	m_origMousePos(),
+	m_displayOffset(0)
+{
+	layoutSetup();
+}
+
+
+LcdFloatSpinBox::LcdFloatSpinBox(int numWhole, int numFrac, const QString& style, const QString& name, QWidget* parent) :
+	FloatModelView(new FloatModel(0, 0, 0, 0, nullptr, name, true), this),
+	m_wholeDisplay(numWhole, style, parent, name, false),
+	m_fractionDisplay(numFrac, style, parent, name, true),
+	m_mouseMoving(false),
+	m_intStep(false),
+	m_origMousePos(),
+	m_displayOffset(0)
+{
+	layoutSetup(style);
+}
+
+
+void LcdFloatSpinBox::layoutSetup(const QString &style)
+{
+	// Assemble the LCD parts
+	QHBoxLayout *lcdLayout = new QHBoxLayout();
+
+	m_wholeDisplay.setSeamless(false, true);
+	m_fractionDisplay.setSeamless(true, false);
+
+	lcdLayout->addWidget(&m_wholeDisplay);
+
+	QLabel *dotLabel = new QLabel("", this);
+	QPixmap dotPixmap(embed::getIconPixmap(QString("lcd_" + style + "_dot").toUtf8().constData()));
+	dotLabel->setPixmap(dotPixmap.copy(0, 0, dotPixmap.size().width(), dotPixmap.size().height() / 2));
+	lcdLayout->addWidget(dotLabel);
+
+	lcdLayout->addWidget(&m_fractionDisplay);
+
+	lcdLayout->setContentsMargins(0, 0, 0, 0);
+	lcdLayout->setSpacing(0);
+
+	// Add space for label
+	QVBoxLayout *outerLayout = new QVBoxLayout();
+	outerLayout->addLayout(lcdLayout);
+	outerLayout->addSpacing(9);
+	outerLayout->setContentsMargins(0, 0, 0, 0);
+	outerLayout->setSizeConstraint(QLayout::SetFixedSize);
+	this->setLayout(outerLayout);
+}
+
+
+void LcdFloatSpinBox::update()
+{
+	const int whole = static_cast<int>(model()->value());
+	const float fraction = model()->value() - whole;
+	const int intFraction = fraction * std::pow(10.f, m_fractionDisplay.numDigits());
+	m_wholeDisplay.setValue(whole);
+	m_fractionDisplay.setValue(intFraction);
+
+	QWidget::update();
+}
+
+
+void LcdFloatSpinBox::contextMenuEvent(QContextMenuEvent* event)
+{
+	CaptionMenu contextMenu(model()->displayName());
+	addDefaultActions(&contextMenu);
+	contextMenu.exec(QCursor::pos());
+}
+
+
+void LcdFloatSpinBox::mousePressEvent(QMouseEvent* event)
+{
+	if (event->button() == Qt::LeftButton &&
+		!(event->modifiers() & Qt::ControlModifier) &&
+		event->y() < m_wholeDisplay.cellHeight() + 2)
+	{
+		m_mouseMoving = true;
+		m_origMousePos = event->globalPos();
+
+		AutomatableModel *thisModel = model();
+		if (thisModel)
+		{
+			thisModel->addJournalCheckPoint();
+			thisModel->saveJournallingState(false);
+		}
+	}
+	else
+	{
+		FloatModelView::mousePressEvent(event);
+	}
+}
+
+
+void LcdFloatSpinBox::mouseMoveEvent(QMouseEvent* event)
+{
+	// switch between integer and fractional step based on cursor position
+	if (event->x() < m_wholeDisplay.width()) { m_intStep = true; }
+	else { m_intStep = false; }
+
+	if (m_mouseMoving)
+	{
+		int dy = event->globalY() - m_origMousePos.y();
+		if (gui->mainWindow()->isShiftPressed()) { dy = qBound(-4, dy/4, 4); }
+		if (dy > 1 || dy < -1)
+		{
+			model()->setValue(model()->value() - dy / 2 * getStep());
+			emit manualChange();
+		}
+	}
+}
```

Comment:
Thanks, that makes sense. I follow the development quite loosely, so I did not notice the `LcdSpinBox` received changes since the Microtuner PR was opened. And since the float version was based on a copy, the recent merges did not really bring any of the updates.